lib/web_ui/lib/src/engine/window.dart

```diff
@@ -182,6 +172,9 @@ class EngineWindow extends ui.Window {
         textEditing.channel.handleTextInput(data);
         return;
 
+      case 'flutter/web_test_e2e':
+        _handleWebTestEnd2EndMessage(data);
+        return;
```

Comment:
One day, someone should go through all the platform messages and call `callback` which indicates to the framework that the handling of the message is done.

For now, you may need to do it for this platform message by calling:
```dart
_replyToPlatformMessage(callback, codec.encodeSuccessEnvelope(true));
```
This will let you await on the `channel.invokeMethod()` in the framework side.

Reply:
Filed https://github.com/flutter/flutter/issues/52929 so we don't forget.

Reply:
Done.

Reply:
Thanks for doing this @ferhatb! Last time I tried, it broke things. I'm glad nothing is broken anymore.